src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/ECDsaCng.ImportExport.cs

```diff
@@ -0,0 +1,558 @@
+                    Emit(blob, ref offset, curve.P);
+                    Emit(blob, ref offset, curve.A);
+                    Emit(blob, ref offset, curve.B);
+                    Emit(blob, ref offset, curve.G.X);
+                    Emit(blob, ref offset, curve.G.Y);
+                    Emit(blob, ref offset, curve.N);
+                    Emit(blob, ref offset, curve.H);
+                    Emit(blob, ref offset, curve.S);
+                    Emit(blob, ref offset, parameters.Q.X);
+                    Emit(blob, ref offset, parameters.Q.Y);
+                    if (includePrivate)
+                    {
+                        Emit(blob, ref offset, parameters.D);
+                    }
+
+                    // We better have computed the right allocation size above!
+                    Debug.Assert(offset == blobSize, "offset == blobSize");
+                }
+
+                return ImportFullKeyBlob(blob, includePrivate);
+            }
+        }
+
+        #endregion
+
+        #region Export Parameters
+        /// <summary>
+        ///     Exports the key used by the ECC object into an ECParameters object.
+        /// </summary>
+        public override ECParameters ExportExplicitCurveParameters(bool includePrivateParameters)
+        {
+            return ExportPrimeCurveParameters(this.Key, includePrivateParameters);
+        }
+
+        /// <summary>
+        ///     Exports the key used by the ECC object into an ECParameters object.
+        /// </summary>
+        public override ECParameters ExportNamedCurveParameters(bool includePrivateParameters)
+        {
+            ECParameters ecparams = ExportNamedCurveParameters(this.Key, includePrivateParameters);
+            ecparams.Curve = new ECNamedCurve(new Oid(null, this.Key.GetCurveName()));
+            return ecparams;
+        }
+
+        private static ECParameters ExportNamedCurveParameters(CngKey key, bool includePrivateParameters)
+        {
+            //
+            // We now have a buffer laid out as follows:
+            //     BCRYPT_ECCKEY_BLOB   header
+            //     byte[cbKey]          Qx
+            //     byte[cbKey]          Qy
+            //     -- Private only --
+            //     byte[cbKey]          D
+            //
+            byte[] ecBlob = ExportKeyBlob(key, includePrivateParameters);
+            ECParameters ecParams = new ECParameters();
+
+            KeyBlobMagicNumber magic = (KeyBlobMagicNumber)BitConverter.ToInt32(ecBlob, 0);
+
+            // Check the magic value in the key blob header. If the blob does not have the required magic,
+            // then throw a CryptographicException.
+            CheckMagicValueOfKey(magic, includePrivateParameters);
+
+            unsafe
+            {
+                // Fail-fast if a rogue provider gave us a blob that isn't even the size of the blob header.
+                if (ecBlob.Length < sizeof(BCRYPT_ECCKEY_BLOB))
```

Comment:
OK. It was copy-pasted this way from RSACng.ImportExport.cs (which I don't think I should change as part of this)